### PR TITLE
Chat code

### LINKED ISSUE #13: API + DB model - CHAT CREDENTIALS endpoint
Create API endpoints for CHAT CREDENTIALS resource:

- [x] - [POST] create new code === check in a guest
- [x] - [PUT] let the code expire === check out a guest
- [ ] - **Update documentation in these endpoints!**

> Make sure, the code is always **unique** and **safely stored!**

**Feature Name:**
Ability to generate unique guest login codes on check in and deactivate on check out

- **User access:**
  - Receptionist
  - Admin
  - Super Admin
- **User Stories:**
  - As a Guest, I want to not have to register for the app so that I can use it quickly.
  - As a Receptionist, I want to quickly generate guest login codes so that I can quickly check in a guest.
  - As a Receptionist, I want to be able to quickly edit the expiration date in case the guest decides to stay at our hotel for a longer period of time.
  - As a Receptionist, I want to be able to quickly terminate the access for the guest to the chat app on his check-out.
- **What internal APIs will you build to implement this feature?**
  - [POST] create new code
  - [PUT] edit expiration date (either to prolong it or let it expire)

### PR DESCRIPTION
**Pull request by Pav0l**
# Description

Please include a summary of the change and a link to which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #13 

- Updated POST requests for `/api/auth/register` to create Super Admin, and `/api/users/` to create Admin/Receptionist/Guest.
- Created a function to generate 10 character `passcode` for guests to use in chat
- Updated email validation function
- Updated tests
- Created Chat login endpoint
- Created Chat delete endpoint (updates the User with `passcode: null, is_left: true`
- Updated GET user by ID to respond with user if _id is for user, or list of hotel users, if _id is hotel_id

To do:
- Update documentation
- Write tests
Will do in a separate PR.

## Type of change

Please add a relevant label.

## Change status

- [x] Complete, but not tested (may need new tests)

# How Has This Been Tested?

Please describe steps taken to ensure this feature is functional and does not break other features:

Tested with debugger. Functionality works
Run Jest tests as I was updating.
Need to create more tests for new endpoints. Will do in separate PR.

# Checklist:

- [x] My code follows the style guidelines of this project

- [x] I have performed a self-review of my own code

- [x] I have commented my code, particularly in hard-to-understand areas

- [ ] I have made corresponding changes to the documentation, if relevant

- [x] I have run the app using my feature and ensured that no functionality is broken

- [x] My changes generate no new warnings

- [x] There are no merge conflicts

### CODE REVIEW

File: models/usersModels.js [hunk 1]
```diff
@@ -8,7 +8,14 @@ const userSchema = new mongoose.Schema({
   hotel_id: {
     type: mongoose.Schema.Types.ObjectId,
     ref: MODEL_NAMES.HOTELS,
-    required: true,
```
Comment (ConConRob) [inline models/usersModels.js:11]: Why don't we want a guest to have a hotel id?

File: routes/auth.js [hunk 2]
```diff
@@ -4,71 +4,68 @@ const createToken = require('../utils/createToken');
...
 
-routes.post(
-  '/login',
-  // add validation middleware
-  async (req, res, next) => {
-    try {
-      const { name, password } = req.body;
+routes.post('/login', async (req, res, next) => {
+  try {
+    const { name, password } = req.body;
```
Comment (ConConRob) [inline routes/auth.js]: Should we change this to email?

**Review by ConConRob — COMMENTED** [review]
Looks really good. I think we want employees to login with their email though

File: models/usersModels.js (hunk 1 above)
Comment (Pav0l) [reply models/usersModels.js:11]: Guests will have `hotel_id`. 
This is for Super Admins, as they will not have `hotel_id` when their account will be created with `POST` request to `/api/auth/register` endpoint.

File: routes/auth.js (hunk 2 above)
Comment (Pav0l) [reply routes/auth.js]: done, thanks for pointing it out! 🔥

**Review by ConConRob — APPROVED** [review]
Looks good, reviewed as a team

**Review by mark-marshall — APPROVED** [review]
🎯

**Pull request merged**